src/OrbitBase/include/OrbitBase/StopSource.h

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef ORBIT_BASE_STOP_SOURCE_H_
+#define ORBIT_BASE_STOP_SOURCE_H_
+
+#include <memory>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/StopState.h"
+#include "OrbitBase/StopToken.h"
+
+namespace orbit_base {
+
+// StopSource together with StopToken is designed to enable thread safe task cancellation. Create a
+// StopSource in the thread that starts a new thread and give that new thread a StopToken (from
```

Comment:
"Create a StopSource when creating a new *task* and give that new *task* a StopToken"? The creation of a task doesn't necessarily imply the creation of a thread, we can be using a thread pool of existing threads for example, or a task could be run on the same thread.

And later: "Then a stop can be requested from any thread by calling StopSource::RequestStop"

Reply:
done